filter.go

```diff
@@ -0,0 +1,86 @@
+package inflight
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type FQFilter struct {
+	// list-watching API models
+	lock   *sync.Mutex
+	queues []*Queue
+
+	*sharedDispatcher
+
+	Matcher  func(*http.Request, []*Queue) *Queue
+	Delegate http.HandlerFunc
+}
+
+func (f *FQFilter) Serve(resp http.ResponseWriter, req *http.Request) {
+
+	matchedQueue := f.Matcher(req, f.queues)
+
+	dispatcher := f.sharedDispatcher.producers[matchedQueue.Priority]
+	func() {
+		dispatcher.lock.Lock()
+		defer dispatcher.lock.Unlock()
+		if dispatcher.requestsexecuting > dispatcher.ACV {
+			// too many requests
+			fmt.Println("throttled...")
+			resp.WriteHeader(http.StatusConflict)
+			return
+		}
+	}()
+	distributionCh := dispatcher.fqScheduler.Enqueue(matchedQueue)
```

Comment:
the enqueued packets and the `requestsexecuting` counter can be leaking when the client requests truncates/aborts here i presume. (i'll run this locally to verify this out).